src/compiler/transformers/declarations.ts

```diff
@@ -617,6 +617,11 @@ export function transformDeclarations(context: TransformationContext) {
 
         function mapReferencesIntoArray(references: FileReference[], outputFilePath: string): (file: SourceFile) => void {
             return file => {
+                if (exportedModulesFromDeclarationEmit?.includes(file.symbol)) {

```

Comment:
This is not what this array was for, and the name confuses me (these aren’t exported from this file, they’re imported 🤔), but it appears to be _exactly_ what I was going to add, so I opted not to duplicate it.